text/0308-cli-advisories.md

```diff
@@ -0,0 +1,263 @@
+
+### Why should I use this feature?
+
+These advisories shown by the CLI contain very important and actionable
+information about problems that directly affect you. They give you an
+opportunity to upgrade the CLI or the construct library when necessary or to
+work around high-impacting issues.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+In case of emergency announcements, such as security vulnerabilities or
+regressions, the only mechanisms we have right now are email campaigns and
+pinned GitHub issues. These are not always the best means to reach out to
+customers. Many email addresses are not monitored by anyone and customers don’t
+necessarily check the GitHub repository for updates. The output of the CLI, on
+the other hand, is seen by many customers every day.
+
+### Why should we *not* do this?
+
+This is a powerful feature to convey urgent and important messages to customers.
+However, to keep its relevance, it should be used sparingly and the messages
+must meet a high bar. Otherwise it will just become a tool for spamming
+customers. We will introduce filters to reduce this risk, by only showing
+content that applies to each particular environment. But ultimately, it hinges
+on responsible use. In particular, this feature should not be used for things
+like marketing campaigns, blog post announcements or upcoming events. If we
+don’t have a good mechanism for defining and enforcing this constraint, we
+should not implement this feature.
+
+### What is the technical solution (design) of this feature?
+
+To publish a new advisory, all that the CDK team will have to do is create a
+GitHub issue that fulfils some requirements. The CLI will consume these issues
+using the GitHub API and apply a set of filters to narrow down the list of
+advisories to the context in which it is being run. To improve the publishing
+experience, we will implement a GitHub action that will validate whether the
+issues that are candidates for advisories fulfil the requirements. For a more
+detailed explanation, see the Appendix.
+
+### Is this a breaking change?
+
+No.
+
+### What alternative solutions did you consider?
+
+All the solutions listed below were considered and discarded for adding
+unnecessary complexity to the overall architecture.
+
+On the publishing side:
+
+* Implementing a new internal REST service to manage the advisories.
+* Storing the advisories as structured data (e.g., JSON files) in some GitHub
+  repository.
+
+On the distribution side:
+
+* Assuming the advisories were stored in an S3 file (using one of the publishing
+  solutions considered above), we would use CloudFront to distribute them.
+* Use GitHub pages as our distribution mechanism, directly from the repository
+  were the data were stored.
+
+### What is the high-level project plan?
+
+1. Implement the issue validation logic as an NPM package.
+2. Implement the GitHub action, using the validation package.
```

Comment:
What's the benefit of publishing this as a separate npm + GitHub Action? Feels like a simple GitHub workflow in the aws-cdk repo with some internal logic should be sufficient? (and less moving parts -- consider each public artifact like that an API we must maintain).